src/shared/infra/http/controllers/wallet/wallet.controller.e2e.spec.ts

```diff
@@ -0,0 +1,96 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import {
+  INestApplication,
+  HttpStatus,
+  NotFoundException,
+} from '@nestjs/common';
+import * as request from 'supertest';
+
+import { AppModule } from '@/app.module';
+import { SaveWalletDTO } from '@/wallet/dto';
+import { AccountType } from '@/shared/constants/enums';
+import { IWalletRepository } from '@/wallet/interfaces';
+
+describe('Wallet Controller E2E', () => {
+  let app: INestApplication;
+  let walletRepository: IWalletRepository;
+  let createWalletMock: jest.SpyInstance;
+
+  const validSaveWalletDTO: SaveWalletDTO = {
+    user_id: 'auth0|58vfb567d5asdea52bc65ebba',
+    bank_id: 'd344a168-60ad-48fc-9d57-64b412e4f6d4',
+    account_type: AccountType.CHECKING_ACCOUNT,
+    description: 'Wallet description',
+  };
+
+  beforeAll(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [AppModule],
+    })
+      .overrideProvider(IWalletRepository)
+      .useValue({
+        create: jest.fn(),
+      })
+      .compile();
+
+    app = module.createNestApplication();
+    walletRepository = module.get<IWalletRepository>(IWalletRepository);
+
+    createWalletMock = jest.spyOn(walletRepository, 'create');
+
+    await app.init();
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  describe('/wallet (POST)', () => {
+    it('should create a wallet', async () => {
+      createWalletMock.mockResolvedValue(validSaveWalletDTO);
+
+      const response = await request(app.getHttpServer())
+        .post('/wallet')
+        .send(validSaveWalletDTO)
+        .expect(HttpStatus.CREATED);
+
+      expect(response.body).toHaveProperty('user_id');
+      expect(response.body).toHaveProperty('bank_id');
+      expect(response.body.account_type).toBe(AccountType.CHECKING_ACCOUNT);
+      expect(response.body.description).toBe('Wallet description');
+    });
+
+    it('should not be able to register a wallet if bank does not exist', async () => {
+      createWalletMock.mockRejectedValue(new NotFoundException());
+
+      const dtoWithNonExistingBank: SaveWalletDTO = {
+        ...validSaveWalletDTO,
+        bank_id: 'd344a168-60ad-48fc-9d57-64b412e4f6d5',
```

Comment:
Coloca "non_existing_bank_id" para ser mais descritivo.

Reply:
Kayke eu tentei fazer isso quando eu tava fazendo os testes, mas não é possível mesmo, o erro not found só é lançado quando tem a quantidade exata de carácteres na requisição, por isso estou mantendo um exemplo real ok?

Reply:
@shimarrudz e se você mockar o find-by-id no FindBankByIdUseCase para ele retornar null? Ai você conseguiria passar um id qualquer sabendo que o repository simularia uma busca que não existiria

Reply:
Pensou bem Henrique, tentei aqui mas nem assim está indo, so quando retorna um id errado mesmo. Mas valeu tentar!